vars/execDeployPipeline.groovy

```diff
@@ -0,0 +1,191 @@
+}
+
+
+private def getJobParams(envs, svcs) {
+    // Set up the job parameters
+    p = []
+    svcs.each { key, data ->
+        def paramName = getParamNameForSvcKey(key, data)
+        def displayName = data.get('displayName', "${key.toString()}")
+        p.add([
+            $class: 'BooleanParameterDefinition',
+            name: paramName, defaultValue: true, description: "Deploy/promote ${displayName}"
+        ])
+    }
+
+    choices = []
+    envs.each { key, data ->
+        choices.add(data['env'])
+    }
+    p.add([
+        $class: 'ChoiceParameterDefinition',
+        name: 'ENV', choices: choices, description: 'The target environment'
+    ])
+    p.add([
+        $class: 'BooleanParameterDefinition',
+        name: 'RELOAD',
+        defaultValue: false,
+        description: "Do nothing, simply re-load this job's groovy file"
+    ])
+
+    return p
+}
+
+
+// Parse the selected parameters when the job is run
+private def parseParams(envs, svcs) {
+    imagesToCopy = []
+    echo "Job params: ${params.toString()}"
+    servicesToSkip = envs[params.ENV].get('skip', [])
+
+    svcs.each { key, data ->
+        paramName = getParamNameForSvcKey(key, data)
+        echo "Checking if ${paramName} is checked and should be deployed..."
+        boxChecked = params.get(paramName.toString())
+        promoteImageOnly = data.get('promoteImageOnly')
+        disableImageCopy = data.get('disableImageCopy')
+        copyImages = envs[params.ENV]['copyImages']
+        deployServices = envs[params.ENV]['deployServices']
+
+        echo(
+            "${key} boxChecked: ${boxChecked}, promoteImageOnly: ${promoteImageOnly}, " +
+            "disableImageCopy: ${disableImageCopy}"
+        )
+
+        // if the service was checked, add its image to the list of images we will copy
+        if (copyImages && !disableImageCopy && boxChecked) imagesToCopy.add(data['srcImage'])
+
+        // if a service was not checked, add it to the list of services to skip, but only
+        // if 'promoteImageOnly' is false (because in that case deployment doesn't apply
+        // for this component)
+        if (deployServices && !boxChecked && !promoteImageOnly) {
+            servicesToSkip.add(data['templateName'])
+        }
+    }
+
+    return [
+        envConfig: envs[params.ENV],
+        imagesToCopy: imagesToCopy,
+        servicesToSkip: servicesToSkip,
+        deployServices: envs[params.ENV]['deployServices']
+    ]
+}
+
+
+def runDeploy(parsed) {
+    // Run the actual deploy after the parameters have been parsed
+    def imagesToCopy = parsed['imagesToCopy']
+    def servicesToSkip = parsed['servicesToSkip']
+    def envConfig = parsed['envConfig']
+    def deployServices = parsed['deployServices']
+
+    echo "imagesToCopy:   ${imagesToCopy}, servicesToSkip: ${servicesToSkip}"
+    echo "envConfig:      ${envConfig}"
+    currentBuild.description = "env: ${envConfig['env']}"
+
+    openShift.withNode {
```

Comment:
Should it be changed to `openShiftUtils.withNode`?

Reply:
Yep. I haven't tested the deploy pipeline jobs yet -- I'm sure I would have ran into this soon Thanks for pointing this out.